File: lib/curl/test/webkit-inspector.cpp
/*
 * =====================================================================================
 *
 *       Filename:  webkit-inspector.cpp
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  02/12/2014 09:28:23 AM
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  Neophytos Demetriou (), 
 *   Organization:  
 *
 * =====================================================================================
 */

#include <QtGui/QApplication>
#include <QtWebKit/QWebInspector>
#include <QtWebKit/QGraphicsWebView>
 
#include "html5applicationviewer.h"
 
int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
 
    Html5ApplicationViewer viewer;
    viewer.setOrientation(Html5ApplicationViewer::ScreenOrientationAuto);
    viewer.showExpanded();
 
    viewer.webView()->page()->settings()->setAttribute(QWebSettings::DeveloperExtrasEnabled, true);
 
    QWebInspector inspector;
    inspector.setPage(viewer.webView()->page());
    inspector.setVisible(true);
 
    viewer.loadFile(QLatin1String("html/index.html"));
 
 
    return app.exec();
} 

File: lib/curl/test/webkit-inspector.py
import sys
from PyQt4.QtCore import *
from PyQt4.QtGui import *
from PyQt4.QtWebKit import *

app = QApplication(sys.argv)

web = QWebView()
# web.settings().setAttribute(QWebSettings.WebAttribute.DeveloperExtrasEnabled, True)
# or globally:
# QWebSettings.globalSettings().setAttribute(
#     QWebSettings.WebAttribute.DeveloperExtrasEnabled, True)

web.load(QUrl("http://www.google.com"))
web.show()

inspect = QWebInspector()
inspect.setPage(web.page())
inspect.show()

sys.exit(app.exec_())
